catalog/task/push-snapshot/0.5/push-snapshot.yaml

```diff
@@ -53,14 +73,21 @@ spec:
             "docker://${data[repository]}:${data[tag]}" 2>/dev/null || true)
           if [[ "$destination_digest" != "$source_digest" || -z "$destination_digest" ]]
           then
-            printf '* Pushing component: %s\n' "${data[name]}"
-            skopeo copy \
-              --all \
-              --preserve-digests \
-              --dest-precompute-digests \
-              --retry-times="$(params.retries)" \
-              "docker://${data[containerImage]}" \
-              "docker://${data[repository]}:${data[tag]}" ;
+            push_image "${data[name]}" "${data[containerImage]}" "${data[repository]}" "${data[tag]}"
+            if [ $(params.addTimestampTag) = true ] ; then
+              timestamp="TODO"
+              push_image "${data[name]}" "${data[containerImage]}" "${data[repository]}" "$timestamp"
+            fi
+            if [ $(params.addShaTag) = true ] ; then
+              if [[ "${data[containerImage]}" == *"@sha256"* && \
+                $(echo "${data[containerImage]}" | tr -cd ':' | wc -c) -eq 1 ]]
+              then
+                sha=$(echo "${data[containerImage]}" | cut -d ':' -f 2)
+                push_image "${data[name]}" "${data[containerImage]}" "${data[repository]}" "$sha"
+              else
+                printf 'Asked to create sha based tag, but no sha found in %s\n' "${data[containerImage]}"
```

Comment:
should this surface as an error?

Reply:
I can error out in this case if you think that makes the most sense. Not sure if we have a snapshot of X images and 1 is bad, we still want to push the X-1 or not

Reply:
I think for the time being, it is better to error out. @davidmogar ?

Reply:
@davidmogar thoughts? One thing is the images prior to the one with the potential issue (if any) would already have been pushed, `exit 1` or not

Reply:
We've been talking about making our pipelines more robust and part of it was to recover from an error on the next run. If this is an error, it should exit with 1 and be recovered in the next run.

Reply:
Added `exit 1`